docs/proposals/deployment-insanity.md

```diff
@@ -0,0 +1,29 @@
+Dropping into a doc so we have a place to argue.
+
+We want to enabling deployments(Ds) and someday we want feature parity with DCs and then we'll likely retire DCs.
+
+This means that we need to make sure that no D or DC have conflicting names.  The simplest way to 
+do this is to have the two resources cohabitate (live in the same directory in etcd).  However,
+they are different than our other migrating/cohabitating resources like replicationcontrollers and 
+replicasets.  RC and RS controllers are close enough to parity that we're going to be able to run
+a single controller to manage both.  D and DC controllers are not close enough to do that.
+
+Since we'll be running two controllers, we'll need to have the controllers know which resources under
+the D or DC endpoint they should be managing.  We don't yet have controllerRefs and we're unlikely to 
+pick them in the next couple days.  The fastest way that I see to do this is:
+ 1. Extend `StorageFactory` to allow the storage encoding version for cohabitating resources vary based
```

Comment:
Why can't we just use `resourceEncodingConfig.SetResourceEncoding` and then pass the config in the factory?
https://github.com/openshift/origin/pull/8274/commits/4df0f42b4d1e951ef81742a683f94c58c83b83e0